pkg/worker/workerhandler/claimresolvehandler/custom_ensure.go

```diff
@@ -17,38 +19,83 @@ import (
 func (h *SystemHandler) Ensure(tas *task.Task, bud *budget.Budget) error {
 	var err error
 
-	//     find claims with lifecycle "propose" that are expired
-	var cla []*poststorage.Object
+	// find trees with lifecycle "propose" that are expired
+	var trees []*poststorage.Object
 	{
-		cla, err = h.sto.Post().SearchExpiry()
+		trees, err = h.sto.Post().SearchExpiry()
 		if err != nil {
 			return tracer.Mask(err)
 		}
 	}
 
-	for _, x := range cla {
-		var tre *big.Int
+	for _, x := range trees {
+		var treeId *big.Int
 		{
-			tre, err = metToTre(x.Meta)
+			treeId, err = metaToTreeId(x.Meta)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		var det [4]marketscontract.IMarketsClaim
+		var claims [4]marketscontract.IMarketsClaim
 		{
-			det, err = h.mar.Claims(nil, tre)
+			claims, err = h.markets.Claims(nil, treeId)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		fmt.Printf("det %#v\n", det)
+		var claimsLength *big.Int
+		{
+			claimsLength, err = h.markets.ClaimsLength(nil, treeId)
+			if err != nil {
+				return tracer.Mask(err)
+			}
 
-		// read from chain if claim has to be resolved, just double checking
+			// sanity check - we cap the number of claims to 4 in the contract
+			if claimsLength.Cmp(big.NewInt(4)) == 1 {
+				return tracer.Maskf(runtime.ExecutionFailedError, "claimresolverhandler: invalid state: too many claims")
```

Comment:
As discussed if we return errors from worker handlers where conditions are never changing, then we create some sort of dead lock in which the task never completes reconciling. If we want to handle certain edge cases, then we could maybe just add some logging. Below some example code.

```go
		w.log.Log(
			context.Background(),
			"level", "info",
			"message", "worker searching for tasks",
			"addr", w.res.Listen(),
		)
```

Reply:
https://github.com/uvio-network/apiserver/pull/49/commits/ecc31442e934f4d10335c15ca228c58b9fdde760